nan.h

```diff
@@ -1527,26 +1755,28 @@ class NanObjectWrap : public node::ObjectWrap {
 template <typename T, typename P>
 class NanWeakCallbackData {
  public:  // constructors
+  inline ~NanWeakCallbackData();
+
+ private:  // constructors
+  NAN_DISALLOW_ASSIGN_COPY_MOVE(NanWeakCallbackData)
+
+ public:
   typedef void (*Callback)(
       NanWeakCallbackData & data  // NOLINT(runtime/references)
       );
-  NanWeakCallbackData(v8::Handle<T> handle, P* param, Callback cb)
-    : parameter(param), callback(cb) {
-    NanAssignPersistent(persistent, handle);
-    Revive();
+  NanWeakCallbackData(v8::Persistent<T> *handle, P* param, Callback cb)
+    : persistent(), parameter(param), callback(cb) {
+    std::memcpy(&persistent, handle, sizeof(persistent));
```

Comment:
Will you lose sleep because of its hackish nature or is it acceptable for lack of better choices?
Maybe the V8 team could change V8 to dispose the persistent for you in the 4.5 WeakCallbackInfo-style weak callbacks? Having this copy of the Persistent handle around is only needed to dispose of it in the weak callback. Nothing else. If V8 took care of that little thing, there would be no need to copy the handle in the future.